Dialog/SpritesEditorDialog.cpp

```diff
@@ -341,6 +360,107 @@ LevelComponents::EntitySet *SpritesEditorDialog::GetCurEntitySetPtr(bool createN
     return curEntityset;
 }
 
+/// <summary>
+/// Render OAM Set based on the ListView data
+/// </summary>
+/// <param name="selectrow">
+/// select a row after re-render oam graphic, leave it blank to set default value 0
+/// </param>
+void SpritesEditorDialog::RenderOamSet(int selectrow)
+{
+    int rowcount = ListViewItemModel->rowCount();
+    if (!rowcount) return;
+    unsigned short *nakedOAMData = new unsigned short[rowcount * 3];
+    for (int i = 0; i < rowcount; ++i)
+    {
+        QStringList CurloadtableStrData = ListViewItemModel->item(i)->text().split(QChar(' '), Qt::SkipEmptyParts);
+        nakedOAMData[3 * i] = CurloadtableStrData.at(0).toUInt(nullptr, 16);
+        nakedOAMData[3 * i + 1] = CurloadtableStrData.at(1).toUInt(nullptr, 16);
+        nakedOAMData[3 * i + 2] = CurloadtableStrData.at(2).toUInt(nullptr, 16);
+    }
+    QPixmap oampixmap(1024, 512);
+    QPainter oampainter(&oampixmap);
+    oampainter.drawImage(0, 0, GetCurEntityPtr()->TestRenderOams(rowcount, nakedOAMData, ui->checkBox_NoReferBox->isChecked()));
+    OAMDesignerMAPScene->clear();
+    OAMmapping = OAMDesignerMAPScene->addPixmap(oampixmap);
+    delete[] nakedOAMData;
+
+    // try to select a row in the listview
+    QModelIndex index = ListViewItemModel->index(selectrow, 0);
+    if (index.isValid())
+        ui->listView_OamDataList->selectionModel()->select(index, QItemSelectionModel::Select);
+    SelectedRow_ListView = selectrow;
+    handleSelectionChanged();
+}
+
+/// <summary>
+/// Helper function used to generate OAM String based on the oam param widgets
+/// </summary>
+/// <return>
+/// single OAM data QString
+/// </return>
+QString SpritesEditorDialog::GenerateOAMString()
+{
+    unsigned short tmpOAMData[3] = {0, 0, 0};
+    int shapesizedata = ui->comboBox_OamShapeType->currentIndex();
+    int yvalue = (ui->spinBox_OamY->value() >= 0) ? ui->spinBox_OamY->value() : 0x100 + ui->spinBox_OamY->value();
+    tmpOAMData[0] = yvalue |
+            (shapesizedata & 0xC) >> 2 << 14 |
```

Comment:
Can you add an extra set of parenthesis here to make the order of operations more clear between >> and <<

Reply:
done.